### ISSUE TITLE
Scenario app is still very flaky on CI

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#145988]
Possible leads:

- [ ] Try reproducing on a Linux emulator:

    ```sh
    # Setup and run Android 34 emulator.
    # ...

    # Run indefinitely until it fails.
    $ testing/scenario_app/tool/deflake_android_tests.sh
    ```

- [x] Consider adding `adb screenrecord` to get visual output of what is failing.
https://github.com/flutter/engine/pull/51832

- [ ] Look into the emulator configuration in CI (`android_34_google_apis_x64.textpb`) for ideas.

- [x] Do not do output file comparisons when the test will fail anyway (gives false negatives)
https://github.com/flutter/engine/pull/51814

---

For example, but also [full list](https://ci.chromium.org/ui/p/flutter/builders/luci.flutter.prod/Linux%20linux_android_emulator_skia_tests):

![image](https://github.com/flutter/flutter/assets/168174/57bb6c0c-9b7f-4c3e-adcf-6437bfc0a2b8)

Every single one of these failures is due to a timeout that looks something like this:

```txt
03-29 06:41:21.667 [I] TestRunner: started: testRotatedMediaSurface_180(dev.flutter.scenariosui.ExternalTextureTests)
03-29 06:41:22.318 [I] flutter: Loading scenario display_texture
03-29 06:41:22.427 [W] Scenarios: Presenting frame 0
03-29 06:41:23.394 [W] ActivityManager: Unable to start service Intent { act=com.google.android.gms.drive.ApiService.RESET_AFTER_BOOT flg=0x4 cmp=com.google.android.gms/.drive.api.ApiService (has extras) } U=0: not found
03-29 06:41:24.177 [W] System: ClassLoader referenced unknown path: 
03-29 06:41:24.250 [W] WindowOnBackDispatcher: sendCancelIfRunning: isInProgress=falsecallback=android.view.ViewRootImpl$$ExternalSyntheticLambda17@cbeae07
03-29 06:41:24.782 [I] Choreographer: Skipped 34 frames!  The application may be doing too much work on its main thread.
03-29 06:41:24.903 [W] FlutterJNI: Tried to send a platform message response, but FlutterJNI was detached from native C++. Could not send. Response ID: 5
03-29 06:41:25.183 [I] flutter: Loading scenario display_texture
03-29 06:41:25.575 [W] Scenarios: Presenting frame 0
...
03-29 06:50:18.252 [I] ActivityManager: Killing 1889:com.google.android.apps.youtube.music/u0a147 (adj 955): empty #17
```

The specific test case is not stable, though it's always a `display_texture` scenario that proceeds.

Notably, the full logs also seem to be missing, which is weird:

![Screenshot 2024-03-29 at 9 05 57 AM](https://github.com/flutter/flutter/assets/168174/4d982a5f-40f2-4927-9a27-150b1546b8fa)

**Comment by matanlurey** [comment]
Another set of flakes here:

```txt
03-29 19:01:01.502 [I] TestRunner: started: getBitmap(dev.flutter.scenariosui.GetBitmapTests)
03-29 19:01:01.512 [W] WindowOnBackDispatcher: sendCancelIfRunning: isInProgress=falsecallback=android.view.ViewRootImpl$$ExternalSyntheticLambda17@307d1d2
skia gold comparison succeeded: ExternalTextureTests_testCanvasSurface.png
03-29 19:01:02.046 [I] flutter: Loading scenario get_bitmap
03-29 19:01:03.167 [W] System: ClassLoader referenced unknown path: 
03-29 19:01:03.265 [W] WindowOnBackDispatcher: sendCancelIfRunning: isInProgress=falsecallback=android.view.ViewRootImpl$$ExternalSyntheticLambda17@d1ea32e
03-29 19:01:03.820 [I] flutter: Loading scenario get_bitmap
03-29 19:01:06.018 [W] System: A resource failed to call Surface.release. 
03-29 19:01:06.022 [W] System: A resource failed to call Surface.release.
```

**Comment by matanlurey** [comment]
Another class of issue is the anti-aliasing inconsistentcy:

![image](https://github.com/flutter/flutter/assets/168174/8483b8e5-29fb-429b-a0a5-512aecef0bc3)

**Comment by matanlurey** [comment]
Ok, looking into recent flakes/failures again:

`test: Android Scenario App Integration Tests (Impeller/Vulkan)` failed (timed out):
https://ci.chromium.org/ui/p/flutter/builders/prod/Linux%20Production%20Engine%20Drone/668409/overview

This one was due to inability to acquire a Swapchain (?):

```txt
04-02 16:05:50.333 [I] TestRunner: finished: testPlatformViewMultiple(dev.flutter.scenariosui.PlatformTextureUiTests)
04-02 16:05:50.337 [I] TestRunner: started: testPlatformViewClippath(dev.flutter.scenariosui.PlatformTextureUiTests)
04-02 16:05:50.558 [I] ActivityManager: Background started FGS: Allowed [callingPackage: com.google.android.inputmethod.latin; callingUid: 10154; uidState: IMPB; uidBFSL: n/a; intent: Intent { cmp=com.google.android.inputmethod.latin/com.google.android.apps.inputmethod.libs.dataservice.superpacks.SuperpacksForegroundTaskService }; code:CURRENT_INPUT_METHOD; tempAllowListReason:<null>; targetSdkVersion:33; callerTargetSdkVersion:33; startForegroundCount:0; bindFromPackage:null: isBindService:false]
04-02 16:05:50.679 [W] ActivityManager: Foreground service started from background can not have location/camera/microphone access: service com.google.android.inputmethod.latin/com.google.android.apps.inputmethod.libs.dataservice.superpacks.SuperpacksForegroundTaskService
04-02 16:05:50.732 [W] ActivityManager: Slow operation: 67ms so far, now at startProcess: returned from zygote!
04-02 16:05:50.802 [W] ActivityManager: Slow operation: 136ms so far, now at startProcess: done updating battery stats
04-02 16:05:50.802 [W] ActivityManager: Slow operation: 136ms so far, now at startProcess: building log message
04-02 16:05:50.802 [I] ActivityManager: Start proc 6155:com.android.emergency/u0a163 for content provider {com.android.emergency/com.android.emergency.EmergencySearchIndexablesProvider}
04-02 16:05:50.802 [W] ActivityManager: Slow operation: 136ms so far, now at startProcess: starting to update pids map
04-02 16:05:50.802 [W] ActivityManager: Slow operation: 136ms so far, now at startProcess: done updating pids map
04-02 16:05:50.895 [I] flutter: [IMPORTANT:flutter/shell/platform/android/android_context_vulkan_impeller.cc(67)] Using the Impeller rendering backend (Vulkan).
04-02 16:05:50.915 [W] ActivityManager: Slow operation: 141ms so far, now at startProcess: done updating battery stats
04-02 16:05:50.915 [W] ActivityManager: Slow operation: 141ms so far, now at startProcess: building log message
04-02 16:05:50.915 [I] ActivityManager: Start proc 6169:com.google.android.apps.messaging/u0a128 for content provider {com.google.android.apps.messaging/com.google.android.apps.messaging.settings.search.MessagesSearchIndexablesProvider}
04-02 16:05:50.915 [W] ActivityManager: Slow operation: 141ms so far, now at startProcess: starting to update pids map
04-02 16:05:50.915 [W] ActivityManager: Slow operation: 141ms so far, now at startProcess: done updating pids map
04-02 16:05:51.136 [W] ActivityManager: Slow operation: 92ms so far, now at startProcess: returned from zygote!
04-02 16:05:51.232 [W] ActivityManager: Slow operation: 188ms so far, now at startProcess: done updating battery stats
04-02 16:05:51.232 [W] ActivityManager: Slow operation: 188ms so far, now at startProcess: building log message
04-02 16:05:51.232 [I] ActivityManager: Start proc 6201:com.google.android.gms.ui/u0a129 for content provider {com.google.android.gms/com.google.android.gms.app.search.GmsSearchIndexablesProvider}
04-02 16:05:51.232 [W] ActivityManager: Slow operation: 188ms so far, now at startProcess: starting to update pids map
04-02 16:05:51.232 [W] ActivityManager: Slow operation: 188ms so far, now at startProcess: done updating pids map
04-02 16:05:51.261 [W] ActivityManager: Slow operation: 184ms so far, now at startProcess: asking zygote to start proc
04-02 16:05:51.300 [W] ActivityManager: Slow operation: 222ms so far, now at startProcess: returned from zygote!
04-02 16:05:51.300 [W] ActivityManager: Slow operation: 222ms so far, now at startProcess: done updating battery stats
04-02 16:05:51.300 [W] ActivityManager: Slow operation: 222ms so far, now at startProcess: building log message
04-02 16:05:51.300 [I] ActivityManager: Start proc 6216:com.google.android.healthconnect.controller/u0a188 for content provider {com.google.android.healthconnect.controller/com.android.healthconnect.controller.searchindexables.HealthConnectSearchIndexablesProvider}
04-02 16:05:51.300 [W] ActivityManager: Slow operation: 222ms so far, now at startProcess: starting to update pids map
04-02 16:05:51.300 [W] ActivityManager: Slow operation: 222ms so far, now at startProcess: done updating pids map
04-02 16:05:51.856 [I] ActivityManager: Flag disabled. Ignoring finishAttachApplication from uid: 10163. pid: 6155
04-02 16:05:52.053 [I] ActivityManager: Flag disabled. Ignoring finishAttachApplication from uid: 10129. pid: 6201
04-02 16:05:52.253 [I] ActivityManager: Flag disabled. Ignoring finishAttachApplication from uid: 10128. pid: 6169
04-02 16:05:52.461 [W] ActivityManager: Slow operation: 202ms so far, now at attachApplicationLocked: before bindApplication
04-02 16:05:52.462 [W] ActivityManager: Slow operation: 203ms so far, now at attachApplicationLocked: immediately before bindApplication
04-02 16:05:52.462 [W] ActivityManager: Slow operation: 204ms so far, now at attachApplicationLocked: immediately after bindApplication
04-02 16:05:52.462 [W] ActivityManager: Slow operation: 204ms so far, now at attachApplicationLocked: after updateLruProcessLocked
04-02 16:05:53.412 [I] ActivityManager: Flag disabled. Ignoring finishAttachApplication from uid: 10188. pid: 6216
04-02 16:05:53.495 [I] Choreographer: Skipped 124 frames!  The application may be doing too much work on its main thread.
04-02 16:05:53.646 [I] flutter: Loading scenario platform_view_clippath
04-02 16:05:53.929 [I] PlatformViewsController: Hosting view in view hierarchy for platform view: 0
04-02 16:05:53.930 [I] PlatformViewsController: PlatformView is using SurfaceProducer backend
04-02 16:05:53.969 [E] OpenGLRenderer: Unable to match the desired swap behavior.
04-02 16:05:53.993 [E] FrameEvents: updateAcquireFence: Did not find frame.
04-02 16:05:54.675 [E] FrameEvents: updateAcquireFence: Did not find frame.
04-02 16:05:55.016 [I] ActivityManager: Background started FGS: Allowed [callingPackage: com.google.android.apps.messaging; callingUid: 10128; uidState: CEM ; uidBFSL: n/a; intent: Intent { cmp=com.google.android.apps.messaging/com.google.apps.tiktok.concurrent.InternalForegroundService (has extras) }; code:BACKGROUND_FGS_PERMISSION; tempAllowListReason:<null>; targetSdkVersion:34; callerTargetSdkVersion:34; startForegroundCount:0; bindFromPackage:null: isBindService:false]
04-02 16:05:55.220 [W] ActivityManager: Foreground service started from background can not have location/camera/microphone access: service com.google.android.apps.messaging/com.google.apps.tiktok.concurrent.InternalForegroundService
04-02 16:05:55.465 [W] vulkan: dequeueBuffer timed out: Connection timed out (-110)
04-02 16:05:55.465 [E] flutter: [ERROR:flutter/impeller/base/validation.cc(59)] Break on 'ImpellerValidationBreak' to inspect point of failure: Could not acquire next swapchain image: Timeout
04-02 16:05:55.465 [E] flutter: [ERROR:flutter/shell/gpu/gpu_surface_vulkan_impeller.cc(64)] No surface available.
```

**Comment by matanlurey** [comment]
@jonahwilliams Seems to think the above is due to "we got into a crashing state, so we crashed".

Unfortunately I can't find anything in the [full logs](https://storage.googleapis.com/flutter_logs/flutter/6883f7313da02a1250c5017189e139a4588d9f2b/Android%20Scenario%20App%20Integration%20Tests%20(Impeller/Vulkan)/5c231f87-6bbb-4865-96c2-2a3dbd41c1d3/run_1.impeller_vulkan.logcat.txt) that would indicate anything, leading me to believe we just entered a terminal state, perhaps due to leaking memory. The hunt continues.
